network_node.go

```diff
@@ -112,37 +114,26 @@ func (n *NetworkNode) GetStream(pid peer.ID) *BasicStream {
 
 func (n *NetworkNode) RefreshStream(pid peer.ID) *BasicStream {
 	// glog.Infof("Creating stream from %v to %v", peer.IDHexEncode(n.Identity), peer.IDHexEncode(pid))
-	if s, ok := n.streams[pid]; ok {
-		s.Stream.Close()
+	if s, ok := n.outStreams[pid]; ok {
+		s.Stream.Reset()
 	}
 
 	ns, err := n.PeerHost.NewStream(context.Background(), pid, Protocol)
+	// glog.Infof("Making new stream to: %v", peer.IDHexEncode(pid))
 	if err != nil {
 		glog.Errorf("%v Error creating stream to %v: %v", peer.IDHexEncode(n.Identity), peer.IDHexEncode(pid), err)
 		return nil
 	}
 	strm := NewBasicStream(ns)
-	n.streams[pid] = strm
-
-	//Handle messages that come back from the stream.  Only do this if we are setting up the Livepeer network protocol.
-	if n.Network != nil {
-		go func() {
-			for {
-				err := streamHandler(n.Network, strm)
-				if err != nil {
-					glog.Errorf("Got error handling stream: %v", err)
-					n.Network.NetworkNode.RemoveStream(strm.Stream.Conn().RemotePeer())
-					strm.Stream.Close()
-					return
-				}
-			}
-		}()
-	}
-
+	n.outStreamsLock.Lock()
+	n.outStreams[pid] = strm
+	n.outStreamsLock.Unlock()
```

Comment:
We remove the for loop here because the new stream is only out-going, so it will never receive messages.